### PR TITLE
Portals - Add configuration warnings for nodes

### PR DESCRIPTION
**Pull request by lawnjelly**
Checks for invalid children / grandchildren etc.

## Notes
* This should be super helpful for users to get their tree in a good structure. Thanks to @akien-mga for explaining how the configuration warnings work.
* I wasn't clear on how often `get_configuration_warning()` gets called, or whether it is editor only, so I've tried to make the checks as efficient as possible.
* In particular it uses a lambda function to do a quick check for incorrect nodes, and does a full check if this flags a problem. I'm not particularly experienced at using lambda functions but it seemed a pretty good fit here, and I can follow a tutorial as well as the next guy. :grin:

### CODE REVIEW

**Comment by Calinou** [comment]
> I wasn't clear on how often get_configuration_warning() gets called, or whether it is editor only, so I've tried to make the checks as efficient as possible.

I think it's editor-only, and it's only called when you call `update_configuration_warning()` or when the scene is loaded.

File: scene/3d/portal.cpp [hunk 1]
```diff
@@ -69,6 +70,40 @@ Portal::~Portal() {
...
 }
 
+String Portal::get_configuration_warning() const {
+	String warning = Spatial::get_configuration_warning();
+
+	auto lambda = [](const Node *p_node) {
+		if (Object::cast_to<RoomManager>(p_node) || Object::cast_to<Room>(p_node) || Object::cast_to<RoomGroup>(p_node)) {
+			return true;
+		}
+		return false;
```
Comment (LightningAA) [inline scene/3d/portal.cpp:80]: Could just be:
```suggestion
		return Object::cast_to<RoomManager>(p_node) || Object::cast_to<Room>(p_node) || Object::cast_to<RoomGroup>(p_node);
```

Comment (lawnjelly) [reply scene/3d/portal.cpp:80]: Yes good point, will change tomorrow. :+1:

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**